lib/filterx/filterx-grammar.ym

```diff
@@ -473,36 +473,49 @@ conditional
 	: if					{ $$ = $1; }
 	| if KW_ELSE codeblock
 	  {
-	    $$ = filterx_conditional_add_false_branch((FilterXConditional*)$1, (FilterXConditional*)filterx_conditional_new_codeblock($3));
+            FilterXExpr *tailing_if = filterx_conditional_find_tail($1);
+	    filterx_conditional_set_false_branch(tailing_if, $3);
+	    $$ = $1;
 	  }
 	;
 
 if
 	: KW_IF '(' expr ')' codeblock
 	  {
-	    $$ = filterx_conditional_new_conditional_codeblock($3, $5);
+	    $$ = filterx_conditional_new($3);
+            filterx_conditional_set_true_branch($$, $5);
 	  }
 	| if KW_ELIF '(' expr ')' codeblock
 	  {
-	    $$ = filterx_conditional_add_false_branch((FilterXConditional*)$1, (FilterXConditional*)filterx_conditional_new_conditional_codeblock($4, $6));
+	    /* create new conditional */
+	    FilterXExpr *elif_expr = filterx_conditional_new($4);
+            filterx_conditional_set_true_branch(elif_expr, $6);
+
+	    /* link it into the else branch of the last if */
+	    filterx_conditional_set_false_branch($1, elif_expr);
```

Comment:
We must link to the tailing if.

```diff
-	     filterx_conditional_set_false_branch($1, elif_expr);
+            FilterXExpr *tailing_if = filterx_conditional_find_tail($1);
+	     filterx_conditional_set_false_branch(tailing_if, elif_expr);
```

Also an E2E test for multiple `elif` branches would be nice.

Reply:
Added a light test and also fixed the bug in the last iteration.